Keep src/engine game agnostic

aleasto opened the pull request:
Move debug lines to engine/
Move all game logic next to main

aleasto committed:
Keep src/engine game agnostic

Move debug lines to engine/
Move all game logic next to main

andreafra commented on src/main.ts:
```diff
@@ -3,11 +3,17 @@ import { utils } from "./utils/utils";
+
+type Mode = "EDITOR" | "GAME";
+
+let editorMode: Mode = "EDITOR";
```
Rename to `mode`

andreafra commented on src/main.ts:
```diff
@@ -65,4 +73,8 @@ async function init() {
 	// );
 }
 
+export function GetEditorMode() {
```
rename to `GetMode()`

andreafra commented on src/main.ts:
```diff
@@ -29,6 +35,8 @@ async function init() {
 	);
 
 	Engine.Setup(gl);
```
Also, since we're 99% using only one webgl context in the future, we might pass the `gl` context in the engine setup step and exporting it from there instead of passing it as a parameter every time :shrug:

andreafra committed:
Refactor camera into class

We now have a map of cameras so we can (technically) switch on the fly between multiple POV.

Updating a camera attribute automatically updates the LookAt matrix

andreafra committed:
Rename editorMode to mode

andreafra committed:
Refactor `gl` out of every function, move `gl` to Core.ts exports

aleasto commented:
lgtm
